{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Archivos y Bases de datos"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import mysql.connector"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "La idea de este taller es manipular archivos (leerlos, parsearlos y escribirlos) y hacer lo mismo con bases de datos estructuradas."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ejercicio 1\n",
    "\n",
    "Baje el archivo de \"All associations with added ontology annotations\" del GWAS Catalog.\n",
    "+ https://www.ebi.ac.uk/gwas/docs/file-downloads\n",
    "\n",
    "Describa las columnas del archivo (_que información estamos mirando? Para qué sirve? Por qué la hicieron?_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Alex\\Anaconda3\\lib\\site-packages\\IPython\\core\\interactiveshell.py:2717: DtypeWarning: Columns (12,23,27) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <th>DATE</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>INTERGENIC</th>\n",
       "      <th>RISK ALLELE FREQUENCY</th>\n",
       "      <th>P-VALUE</th>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <th>OR or BETA</th>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <th>CNV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2009-09-28</td>\n",
       "      <td>18403759</td>\n",
       "      <td>Ober C</td>\n",
       "      <td>2008-04-09</td>\n",
       "      <td>N Engl J Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/18403759</td>\n",
       "      <td>Effect of variation in CHI3L1 on serum YKL-40 ...</td>\n",
       "      <td>YKL-40 levels</td>\n",
       "      <td>632 Hutterite individuals</td>\n",
       "      <td>443 European ancestry cases, 491 European ance...</td>\n",
       "      <td>...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29</td>\n",
       "      <td>1e-13</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>[NR] ng/ml decrease</td>\n",
       "      <td>Affymetrix [290325]</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG  PUBMEDID FIRST AUTHOR        DATE       JOURNAL  \\\n",
       "0            2009-09-28  18403759       Ober C  2008-04-09  N Engl J Med   \n",
       "\n",
       "                                   LINK  \\\n",
       "0  www.ncbi.nlm.nih.gov/pubmed/18403759   \n",
       "\n",
       "                                               STUDY  DISEASE/TRAIT  \\\n",
       "0  Effect of variation in CHI3L1 on serum YKL-40 ...  YKL-40 levels   \n",
       "\n",
       "         INITIAL SAMPLE SIZE  \\\n",
       "0  632 Hutterite individuals   \n",
       "\n",
       "                             REPLICATION SAMPLE SIZE ...   \\\n",
       "0  443 European ancestry cases, 491 European ance... ...    \n",
       "\n",
       "                 CONTEXT INTERGENIC RISK ALLELE FREQUENCY P-VALUE PVALUE_MLOG  \\\n",
       "0  upstream_gene_variant        0.0                  0.29   1e-13        13.0   \n",
       "\n",
       "   P-VALUE (TEXT)  OR or BETA        95% CI (TEXT)  \\\n",
       "0             NaN         0.3  [NR] ng/ml decrease   \n",
       "\n",
       "   PLATFORM [SNPS PASSING QC]  CNV  \n",
       "0         Affymetrix [290325]    N  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df= pd.read_csv('C:/Users/Alex/Documents/eafit/semestres/X semestre/programacion/taller2.tsv', sep = '\\t')\n",
    "df[:1]\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Qué Entidades (tablas) puede definir?"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "1. Enfermedad \n",
    "2. Plataforma (tecnologia de secuenciacion)\n",
    "3. Loci\n",
    "4. Enfermedad-loci\n",
    "5. Journal \n",
    "6. Estudio\n",
    "7. Publicacion "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cree la base de datos (copie el código SQL que se usó)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-3-07d17c338c80>, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-3-07d17c338c80>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    CREATE TABLE enfermedad\u001b[0m\n\u001b[0m               ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "CREATE TABLE enfermedad\n",
    "(\n",
    "id_enfermedad int AUTO_INCREMENT PRIMARY KEY,\n",
    "nombre varchar(255)\n",
    ");\n",
    "\n",
    "create table plataforma\n",
    "(\n",
    "id_plataforma int AUTO_INCREMENT primary key, \n",
    "nombre varchar(255)\n",
    ");\n",
    "\n",
    "CREATE TABLE loci\n",
    "(\n",
    "id_loci int AUTO_INCREMENT PRIMARY KEY,\n",
    "id_plataforma int,\n",
    "foreign key (id_plataforma) references plataforma(id_plataforma),\n",
    "region varchar(255),\n",
    "chrom varchar(255), \n",
    "pos varchar(255),\n",
    "genes_reportados varchar(255),\n",
    "gen_mapped varchar(255),\n",
    "gen_upstream varchar(255),\n",
    "gen_downstream varchar(255),\n",
    "SNP_GENE_IDS varchar(255),\n",
    "UPSTREAM_GENE_DISTANCE varchar(255),\n",
    "DOWNSTREAM_GENE_DISTANCE varchar(255),\n",
    "STRONGEST_SP_RISK varchar(255),\n",
    "SNPS varchar(255),\n",
    "MERGED varchar(255),\n",
    "SNP_ID_CURRENT varchar(255),\n",
    "CONTEXTO varchar(500),\n",
    "risk_allele varchar(255)\n",
    "#PVAl int,\n",
    "#PVALUE_MLOG int,\n",
    "#PVALUE_txt varchar(255),\n",
    "#BETA int,  \n",
    "#novCI varchar(255)\n",
    ");\n",
    "\n",
    "CREATE TABLE enfermedad_loci\n",
    "(\n",
    "id_loci int ,\n",
    "id_enfermedad int,\n",
    "PRIMARY KEY (id_loci, id_enfermedad),\n",
    "foreign key (id_loci) references loci(id_loci),\n",
    "foreign key (id_enfermedad) references enfermedad(id_enfermedad)\n",
    ");\n",
    "\n",
    "CREATE TABLE journal\n",
    "(\n",
    "id_journal int AUTO_INCREMENT PRIMARY KEY,\n",
    "nombre varchar(255)\n",
    ");\n",
    "\n",
    "CREATE TABLE estudio\n",
    "(\n",
    "id_estudio int AUTO_INCREMENT PRIMARY KEY,\n",
    "id_journal int,\n",
    "id_enfermedad int,\n",
    "foreign key (id_journal) references journal(id_journal),\n",
    "foreign key (id_enfermedad) references enfermedad(id_enfermedad),\n",
    "nombre varchar(255),\n",
    "tamano_muestra varchar(500),\n",
    "replicas varchar(500)\n",
    ");\n",
    "\n",
    "create table publicacion\n",
    "(\n",
    "id_publicacion int AUTO_INCREMENT PRIMARY KEY,\n",
    "id_journal int,\n",
    "id_estudio int,\n",
    "foreign key (id_journal) references journal(id_journal),\n",
    "foreign key (id_estudio) references estudio(id_estudio),\n",
    "id_pubmed int,\n",
    "autor varchar (255),\n",
    "fecha_pub varchar (20),\n",
    "link varchar (500)\n",
    ");\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ejercicio 2\n",
    "\n",
    "Lea el archivo y guarde la infomación en la base de datos en las tablas que se definidieron en el __Ejercicio 1__."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <th>DATE</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>INTERGENIC</th>\n",
       "      <th>RISK ALLELE FREQUENCY</th>\n",
       "      <th>P-VALUE</th>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <th>OR or BETA</th>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <th>CNV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2009-09-28</td>\n",
       "      <td>18403759</td>\n",
       "      <td>Ober C</td>\n",
       "      <td>2008-04-09</td>\n",
       "      <td>N Engl J Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/18403759</td>\n",
       "      <td>Effect of variation in CHI3L1 on serum YKL-40 ...</td>\n",
       "      <td>YKL-40 levels</td>\n",
       "      <td>632 Hutterite individuals</td>\n",
       "      <td>443 European ancestry cases, 491 European ance...</td>\n",
       "      <td>...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.29</td>\n",
       "      <td>1e-13</td>\n",
       "      <td>13.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3</td>\n",
       "      <td>[NR] ng/ml decrease</td>\n",
       "      <td>Affymetrix [290325]</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG  PUBMEDID FIRST AUTHOR        DATE       JOURNAL  \\\n",
       "0            2009-09-28  18403759       Ober C  2008-04-09  N Engl J Med   \n",
       "\n",
       "                                   LINK  \\\n",
       "0  www.ncbi.nlm.nih.gov/pubmed/18403759   \n",
       "\n",
       "                                               STUDY  DISEASE/TRAIT  \\\n",
       "0  Effect of variation in CHI3L1 on serum YKL-40 ...  YKL-40 levels   \n",
       "\n",
       "         INITIAL SAMPLE SIZE  \\\n",
       "0  632 Hutterite individuals   \n",
       "\n",
       "                             REPLICATION SAMPLE SIZE ...   \\\n",
       "0  443 European ancestry cases, 491 European ance... ...    \n",
       "\n",
       "                 CONTEXT INTERGENIC RISK ALLELE FREQUENCY P-VALUE PVALUE_MLOG  \\\n",
       "0  upstream_gene_variant        0.0                  0.29   1e-13        13.0   \n",
       "\n",
       "   P-VALUE (TEXT)  OR or BETA        95% CI (TEXT)  \\\n",
       "0             NaN         0.3  [NR] ng/ml decrease   \n",
       "\n",
       "   PLATFORM [SNPS PASSING QC]  CNV  \n",
       "0         Affymetrix [290325]    N  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 22\n",
      "2 23\n",
      "3 24\n",
      "3 25\n",
      "3 26\n",
      "3 27\n",
      "10 28\n",
      "11 29\n",
      "11 30\n",
      "11 31\n",
      "11 32\n",
      "11 33\n",
      "12 34\n",
      "12 35\n",
      "12 36\n",
      "12 37\n",
      "12 38\n",
      "13 39\n",
      "14 40\n",
      "15 41\n",
      "15 42\n",
      "16 43\n",
      "16 44\n",
      "16 45\n",
      "16 46\n",
      "16 47\n",
      "17 48\n",
      "17 49\n",
      "17 50\n",
      "17 51\n",
      "17 52\n",
      "17 53\n",
      "17 54\n",
      "17 55\n",
      "17 56\n",
      "17 57\n",
      "17 58\n",
      "18 59\n",
      "19 60\n",
      "19 61\n",
      "20 62\n",
      "21 63\n",
      "22 64\n",
      "23 65\n",
      "23 66\n",
      "23 67\n",
      "23 68\n",
      "23 69\n",
      "23 70\n",
      "23 71\n",
      "24 72\n",
      "25 73\n",
      "26 74\n",
      "26 75\n",
      "26 76\n",
      "26 77\n",
      "26 78\n",
      "26 79\n",
      "27 80\n",
      "27 81\n",
      "27 82\n",
      "27 83\n",
      "27 84\n",
      "18 85\n",
      "18 86\n",
      "26 87\n",
      "26 88\n",
      "26 89\n",
      "26 90\n",
      "26 91\n",
      "18 92\n",
      "15 93\n",
      "28 94\n",
      "29 95\n",
      "15 96\n",
      "30 97\n",
      "31 98\n",
      "31 99\n",
      "32 100\n",
      "33 101\n",
      "34 102\n",
      "34 103\n",
      "34 104\n",
      "11 105\n",
      "35 106\n",
      "35 107\n",
      "35 108\n",
      "35 109\n",
      "35 110\n",
      "36 111\n",
      "36 112\n",
      "36 113\n",
      "37 114\n",
      "37 115\n",
      "38 116\n",
      "38 117\n",
      "39 118\n",
      "40 119\n",
      "40 120\n",
      "40 121\n",
      "41 122\n",
      "42 123\n",
      "43 124\n",
      "43 125\n",
      "43 126\n",
      "43 127\n",
      "44 128\n",
      "44 129\n",
      "44 130\n",
      "45 131\n",
      "46 132\n",
      "47 133\n",
      "47 134\n",
      "47 135\n",
      "48 136\n",
      "48 137\n",
      "49 138\n",
      "49 139\n",
      "50 140\n",
      "50 141\n",
      "50 142\n",
      "51 143\n",
      "51 144\n",
      "52 145\n",
      "52 146\n",
      "53 147\n",
      "13 148\n",
      "13 149\n",
      "13 150\n",
      "13 151\n",
      "13 152\n",
      "17 153\n",
      "17 154\n",
      "17 155\n",
      "17 156\n",
      "17 157\n",
      "54 158\n",
      "54 159\n",
      "55 160\n",
      "55 161\n",
      "56 162\n",
      "56 163\n",
      "57 164\n",
      "57 165\n",
      "57 166\n",
      "58 167\n",
      "58 168\n",
      "58 169\n",
      "58 170\n",
      "59 171\n",
      "60 172\n",
      "60 173\n",
      "61 174\n",
      "62 175\n",
      "62 176\n",
      "63 177\n",
      "63 178\n",
      "64 179\n",
      "64 180\n",
      "64 181\n",
      "65 182\n",
      "65 183\n",
      "66 184\n",
      "66 185\n",
      "28 186\n",
      "30 187\n",
      "67 188\n",
      "67 189\n",
      "68 190\n",
      "68 191\n",
      "68 192\n",
      "27 193\n",
      "69 194\n",
      "70 195\n",
      "71 196\n",
      "72 197\n",
      "72 198\n",
      "72 199\n",
      "72 200\n",
      "72 201\n",
      "73 202\n",
      "74 203\n",
      "74 204\n",
      "74 205\n",
      "75 206\n",
      "76 207\n",
      "76 208\n",
      "75 209\n",
      "75 210\n",
      "75 211\n",
      "77 212\n",
      "24 213\n",
      "11 214\n",
      "11 215\n",
      "78 216\n",
      "17 217\n",
      "17 218\n",
      "16 219\n",
      "79 220\n",
      "79 221\n",
      "79 222\n",
      "79 223\n",
      "79 224\n",
      "76 225\n",
      "76 226\n",
      "76 227\n",
      "76 228\n",
      "67 229\n",
      "67 230\n",
      "67 231\n",
      "67 232\n",
      "67 233\n",
      "80 234\n",
      "80 235\n",
      "30 236\n",
      "30 237\n",
      "30 238\n",
      "30 239\n",
      "24 240\n",
      "24 241\n",
      "24 242\n",
      "24 243\n",
      "24 244\n",
      "12 245\n",
      "12 246\n",
      "12 247\n",
      "12 248\n",
      "12 249\n",
      "67 250\n",
      "67 251\n",
      "67 252\n",
      "67 253\n",
      "67 254\n",
      "24 255\n",
      "24 256\n",
      "24 257\n",
      "24 258\n",
      "24 259\n",
      "13 260\n",
      "13 261\n",
      "13 262\n",
      "13 263\n",
      "13 264\n",
      "13 265\n",
      "13 266\n",
      "13 267\n",
      "79 268\n",
      "81 269\n",
      "81 270\n",
      "81 271\n",
      "82 272\n",
      "12 273\n",
      "12 274\n",
      "67 275\n",
      "67 276\n",
      "36 277\n",
      "17 278\n",
      "17 279\n",
      "14 280\n",
      "67 281\n",
      "28 282\n",
      "28 283\n",
      "28 284\n",
      "28 285\n",
      "28 286\n",
      "12 287\n",
      "10 288\n",
      "10 289\n",
      "83 290\n",
      "84 291\n",
      "84 292\n",
      "84 293\n",
      "85 294\n",
      "84 295\n",
      "86 296\n",
      "78 297\n",
      "17 298\n",
      "87 299\n",
      "42 300\n",
      "42 301\n",
      "88 302\n",
      "32 303\n",
      "89 304\n",
      "90 305\n",
      "90 306\n",
      "91 307\n",
      "91 308\n",
      "92 309\n",
      "93 310\n",
      "91 311\n",
      "91 312\n",
      "94 313\n",
      "36 314\n",
      "95 315\n",
      "95 316\n",
      "95 317\n",
      "95 318\n",
      "95 319\n",
      "87 320\n",
      "87 321\n",
      "87 322\n",
      "87 323\n",
      "87 324\n",
      "93 325\n",
      "90 326\n",
      "96 327\n",
      "96 328\n",
      "96 329\n",
      "97 330\n",
      "97 331\n",
      "97 332\n",
      "97 333\n",
      "91 334\n",
      "52 335\n",
      "98 336\n",
      "98 337\n",
      "98 338\n",
      "98 339\n",
      "98 340\n",
      "99 341\n",
      "52 342\n",
      "100 343\n",
      "31 344\n",
      "101 345\n",
      "2 346\n",
      "102 347\n",
      "90 348\n",
      "96 349\n",
      "33 350\n",
      "103 351\n",
      "67 352\n",
      "67 353\n",
      "67 354\n",
      "67 355\n",
      "67 356\n",
      "104 357\n",
      "105 358\n",
      "12 359\n",
      "81 360\n",
      "106 361\n",
      "106 362\n",
      "106 363\n",
      "107 364\n",
      "107 365\n",
      "107 366\n",
      "108 367\n",
      "75 368\n",
      "75 369\n",
      "30 370\n",
      "14 371\n",
      "14 372\n",
      "14 373\n",
      "23 374\n",
      "23 375\n",
      "23 376\n",
      "23 377\n",
      "23 378\n",
      "65 379\n",
      "109 380\n",
      "110 381\n",
      "110 382\n",
      "110 383\n",
      "110 384\n",
      "110 385\n",
      "110 386\n",
      "111 387\n",
      "111 388\n",
      "23 389\n",
      "23 390\n",
      "23 391\n",
      "30 392\n",
      "107 393\n",
      "107 394\n",
      "106 395\n",
      "67 396\n",
      "67 397\n",
      "67 398\n",
      "67 399\n",
      "67 400\n",
      "67 401\n",
      "67 402\n",
      "67 403\n",
      "67 404\n",
      "67 405\n",
      "67 406\n",
      "67 407\n",
      "67 408\n",
      "67 409\n",
      "67 410\n",
      "67 411\n",
      "67 412\n",
      "67 413\n",
      "67 414\n",
      "67 415\n",
      "67 416\n",
      "67 417\n",
      "67 418\n",
      "67 419\n",
      "67 420\n",
      "99 421\n",
      "99 422\n",
      "98 423\n",
      "94 424\n",
      "94 425\n",
      "94 426\n",
      "87 427\n",
      "87 428\n",
      "87 429\n",
      "87 430\n",
      "112 431\n",
      "12 432\n",
      "28 433\n",
      "67 434\n",
      "17 435\n",
      "17 436\n",
      "88 437\n",
      "12 438\n",
      "12 439\n",
      "12 440\n",
      "12 441\n",
      "12 442\n",
      "13 443\n",
      "24 444\n",
      "24 445\n",
      "24 446\n",
      "67 447\n",
      "67 448\n",
      "67 449\n",
      "67 450\n",
      "79 451\n",
      "79 452\n",
      "67 453\n",
      "67 454\n",
      "67 455\n",
      "67 456\n",
      "67 457\n",
      "67 458\n",
      "67 459\n",
      "67 460\n",
      "30 461\n",
      "30 462\n",
      "30 463\n",
      "24 464\n",
      "24 465\n",
      "24 466\n",
      "24 467\n",
      "24 468\n",
      "12 469\n",
      "12 470\n",
      "16 471\n",
      "12 472\n",
      "12 473\n",
      "24 474\n",
      "24 475\n",
      "24 476\n",
      "76 477\n",
      "76 478\n",
      "72 479\n",
      "67 480\n",
      "67 481\n",
      "30 482\n",
      "30 483\n",
      "64 484\n",
      "64 485\n",
      "64 486\n",
      "64 487\n",
      "64 488\n",
      "64 489\n",
      "64 490\n",
      "62 491\n",
      "62 492\n",
      "60 493\n",
      "60 494\n",
      "60 495\n",
      "60 496\n",
      "60 497\n",
      "60 498\n",
      "59 499\n",
      "59 500\n",
      "59 501\n",
      "59 502\n",
      "58 503\n",
      "58 504\n",
      "58 505\n",
      "58 506\n",
      "58 507\n",
      "58 508\n",
      "58 509\n",
      "58 510\n",
      "58 511\n",
      "58 512\n",
      "57 513\n",
      "57 514\n",
      "57 515\n",
      "57 516\n",
      "57 517\n",
      "56 518\n",
      "13 519\n",
      "13 520\n",
      "13 521\n",
      "13 522\n",
      "65 523\n",
      "66 524\n",
      "66 525\n",
      "66 526\n",
      "66 527\n",
      "66 528\n",
      "66 529\n",
      "66 530\n",
      "66 531\n",
      "66 532\n",
      "66 533\n",
      "66 534\n",
      "66 535\n",
      "66 536\n",
      "66 537\n",
      "66 538\n",
      "66 539\n",
      "51 540\n",
      "51 541\n",
      "52 542\n",
      "52 543\n",
      "52 544\n",
      "52 545\n",
      "43 546\n",
      "43 547\n",
      "43 548\n",
      "43 549\n",
      "43 550\n",
      "43 551\n",
      "43 552\n",
      "43 553\n",
      "43 554\n",
      "43 555\n",
      "43 556\n",
      "43 557\n",
      "42 558\n",
      "42 559\n",
      "42 560\n",
      "42 561\n",
      "42 562\n",
      "48 563\n",
      "48 564\n",
      "48 565\n",
      "48 566\n",
      "48 567\n",
      "48 568\n",
      "48 569\n",
      "48 570\n",
      "48 571\n",
      "48 572\n",
      "48 573\n",
      "41 574\n",
      "35 575\n",
      "35 576\n",
      "35 577\n",
      "35 578\n",
      "35 579\n",
      "35 580\n",
      "35 581\n",
      "35 582\n",
      "35 583\n",
      "35 584\n",
      "35 585\n",
      "35 586\n",
      "35 587\n",
      "35 588\n",
      "35 589\n",
      "35 590\n",
      "91 591\n",
      "91 592\n",
      "91 593\n",
      "91 594\n",
      "91 595\n",
      "91 596\n",
      "91 597\n",
      "91 598\n",
      "91 599\n",
      "91 600\n",
      "91 601\n",
      "91 602\n",
      "91 603\n",
      "91 604\n",
      "91 605\n",
      "91 606\n",
      "91 607\n",
      "91 608\n",
      "91 609\n",
      "91 610\n",
      "91 611\n",
      "91 612\n",
      "91 613\n",
      "91 614\n",
      "91 615\n",
      "91 616\n",
      "91 617\n",
      "91 618\n",
      "91 619\n",
      "91 620\n",
      "91 621\n",
      "91 622\n",
      "91 623\n",
      "91 624\n",
      "91 625\n",
      "91 626\n",
      "91 627\n",
      "91 628\n",
      "91 629\n",
      "91 630\n",
      "91 631\n",
      "91 632\n",
      "91 633\n",
      "91 634\n",
      "47 635\n",
      "47 636\n",
      "47 637\n",
      "47 638\n",
      "47 639\n",
      "47 640\n",
      "18 641\n",
      "18 642\n",
      "26 643\n",
      "26 644\n",
      "26 645\n",
      "12 646\n",
      "12 647\n",
      "12 648\n",
      "12 649\n",
      "12 650\n",
      "12 651\n",
      "12 652\n",
      "12 653\n",
      "12 654\n",
      "12 655\n",
      "12 656\n",
      "95 657\n",
      "95 658\n",
      "95 659\n",
      "79 660\n",
      "79 661\n",
      "79 662\n",
      "79 663\n",
      "79 664\n",
      "79 665\n",
      "79 666\n",
      "79 667\n",
      "79 668\n",
      "79 669\n",
      "79 670\n",
      "79 671\n",
      "79 672\n",
      "79 673\n",
      "79 674\n",
      "79 675\n",
      "79 676\n",
      "79 677\n",
      "79 678\n",
      "79 679\n",
      "79 680\n",
      "28 681\n",
      "112 682\n",
      "112 683\n",
      "112 684\n",
      "112 685\n",
      "112 686\n",
      "112 687\n",
      "24 688\n",
      "24 689\n",
      "24 690\n",
      "24 691\n",
      "113 692\n",
      "90 693\n",
      "90 694\n",
      "90 695\n",
      "96 696\n",
      "96 697\n",
      "96 698\n",
      "36 699\n",
      "2 700\n",
      "2 701\n",
      "11 702\n",
      "11 703\n",
      "11 704\n",
      "2 705\n",
      "16 706\n",
      "16 707\n",
      "16 708\n",
      "14 709\n",
      "17 710\n",
      "17 711\n",
      "17 712\n",
      "17 713\n",
      "17 714\n",
      "17 715\n",
      "17 716\n",
      "17 717\n",
      "17 718\n",
      "17 719\n",
      "18 720\n",
      "18 721\n",
      "23 722\n",
      "23 723\n",
      "114 724\n",
      "115 725\n",
      "67 726\n",
      "67 727\n",
      "30 728\n",
      "115 729\n",
      "26 730\n",
      "26 731\n",
      "26 732\n",
      "26 733\n",
      "26 734\n",
      "23 735\n",
      "23 736\n",
      "23 737\n",
      "23 738\n",
      "23 739\n",
      "23 740\n",
      "23 741\n",
      "25 742\n",
      "25 743\n",
      "25 744\n",
      "25 745\n",
      "25 746\n",
      "26 747\n",
      "26 748\n",
      "26 749\n",
      "26 750\n",
      "33 751\n",
      "33 752\n",
      "31 753\n",
      "31 754\n",
      "116 755\n",
      "32 756\n",
      "32 757\n",
      "117 758\n",
      "50 759\n",
      "50 760\n",
      "50 761\n",
      "50 762\n",
      "50 763\n",
      "50 764\n",
      "50 765\n",
      "49 766\n",
      "49 767\n",
      "49 768\n",
      "12 769\n",
      "12 770\n",
      "103 771\n",
      "103 772\n",
      "97 773\n",
      "97 774\n",
      "67 775\n",
      "18 776\n",
      "18 777\n",
      "18 778\n",
      "18 779\n",
      "18 780\n",
      "26 781\n",
      "26 782\n",
      "26 783\n",
      "26 784\n",
      "118 785\n",
      "107 786\n",
      "107 787\n",
      "107 788\n",
      "28 789\n",
      "28 790\n",
      "42 791\n",
      "42 792\n",
      "67 793\n",
      "67 794\n",
      "76 795\n",
      "76 796\n",
      "76 797\n",
      "3 798\n",
      "119 799\n",
      "119 800\n",
      "30 801\n",
      "30 802\n",
      "30 803\n",
      "30 804\n",
      "30 805\n",
      "30 806\n",
      "30 807\n",
      "30 808\n",
      "30 809\n",
      "14 810\n",
      "14 811\n",
      "14 812\n",
      "14 813\n",
      "14 814\n",
      "14 815\n",
      "120 816\n",
      "121 817\n",
      "122 818\n",
      "13 819\n",
      "71 820\n",
      "71 821\n",
      "71 822\n",
      "122 823\n",
      "122 824\n",
      "122 825\n",
      "122 826\n",
      "122 827\n",
      "122 828\n",
      "24 829\n",
      "24 830\n",
      "24 831\n",
      "24 832\n",
      "24 833\n",
      "123 834\n",
      "123 835\n",
      "124 836\n",
      "124 837\n",
      "115 838\n",
      "115 839\n",
      "15 840\n",
      "15 841\n",
      "15 842\n",
      "125 843\n",
      "125 844\n",
      "125 845\n",
      "126 846\n",
      "126 847\n",
      "3 848\n",
      "3 849\n",
      "3 850\n",
      "127 851\n",
      "24 852\n",
      "24 853\n",
      "24 854\n",
      "24 855\n",
      "24 856\n",
      "24 857\n",
      "128 858\n",
      "128 859\n",
      "128 860\n",
      "128 861\n",
      "128 862\n",
      "128 863\n",
      "128 864\n",
      "128 865\n",
      "128 866\n",
      "128 867\n",
      "128 868\n",
      "11 869\n",
      "11 870\n",
      "11 871\n",
      "11 872\n",
      "24 873\n",
      "24 874\n",
      "24 875\n",
      "24 876\n",
      "24 877\n",
      "24 878\n",
      "24 879\n",
      "24 880\n",
      "129 881\n",
      "129 882\n",
      "129 883\n",
      "129 884\n",
      "27 885\n",
      "72 886\n",
      "72 887\n",
      "72 888\n",
      "72 889\n",
      "72 890\n",
      "72 891\n",
      "72 892\n",
      "72 893\n",
      "72 894\n",
      "130 895\n",
      "130 896\n",
      "130 897\n",
      "130 898\n",
      "130 899\n",
      "130 900\n",
      "131 901\n",
      "72 902\n",
      "28 903\n",
      "18 904\n",
      "18 905\n",
      "132 906\n",
      "132 907\n",
      "132 908\n",
      "132 909\n",
      "132 910\n",
      "115 911\n",
      "115 912\n",
      "3 913\n",
      "3 914\n",
      "72 915\n",
      "133 916\n",
      "52 917\n",
      "134 918\n",
      "134 919\n",
      "134 920\n",
      "134 921\n",
      "134 922\n",
      "134 923\n",
      "134 924\n",
      "134 925\n",
      "134 926\n",
      "134 927\n",
      "134 928\n",
      "134 929\n",
      "134 930\n",
      "134 931\n",
      "134 932\n",
      "134 933\n",
      "134 934\n",
      "134 935\n",
      "134 936\n",
      "134 937\n",
      "134 938\n",
      "134 939\n",
      "134 940\n",
      "134 941\n",
      "134 942\n",
      "134 943\n",
      "135 944\n",
      "135 945\n",
      "135 946\n",
      "135 947\n",
      "135 948\n",
      "135 949\n",
      "136 950\n",
      "136 951\n",
      "136 952\n",
      "136 953\n",
      "27 954\n",
      "137 955\n",
      "137 956\n",
      "137 957\n",
      "137 958\n",
      "137 959\n",
      "137 960\n",
      "137 961\n",
      "137 962\n",
      "137 963\n",
      "137 964\n",
      "137 965\n",
      "137 966\n",
      "137 967\n",
      "137 968\n",
      "137 969\n",
      "137 970\n",
      "36 971\n",
      "36 972\n",
      "36 973\n",
      "36 974\n",
      "36 975\n",
      "36 976\n",
      "36 977\n",
      "36 978\n",
      "36 979\n",
      "36 980\n",
      "36 981\n",
      "138 982\n",
      "138 983\n",
      "138 984\n",
      "138 985\n",
      "138 986\n",
      "138 987\n",
      "138 988\n",
      "138 989\n",
      "138 990\n",
      "138 991\n",
      "138 992\n",
      "138 993\n",
      "138 994\n",
      "138 995\n",
      "138 996\n",
      "26 997\n",
      "26 998\n",
      "26 999\n",
      "26 1000\n",
      "26 1001\n",
      "136 1002\n",
      "25 1003\n",
      "25 1004\n",
      "18 1005\n",
      "18 1006\n",
      "12 1007\n",
      "12 1008\n",
      "26 1009\n",
      "26 1010\n",
      "26 1011\n",
      "18 1012\n",
      "18 1013\n",
      "18 1014\n",
      "18 1015\n",
      "18 1016\n",
      "18 1017\n",
      "18 1018\n",
      "18 1019\n",
      "18 1020\n",
      "18 1021\n",
      "25 1022\n",
      "25 1023\n",
      "25 1024\n",
      "25 1025\n",
      "25 1026\n",
      "25 1027\n",
      "25 1028\n",
      "25 1029\n",
      "136 1030\n",
      "136 1031\n",
      "136 1032\n",
      "136 1033\n",
      "136 1034\n",
      "136 1035\n",
      "136 1036\n",
      "136 1037\n",
      "136 1038\n",
      "136 1039\n",
      "25 1040\n",
      "25 1041\n",
      "25 1042\n",
      "25 1043\n",
      "18 1044\n",
      "18 1045\n",
      "18 1046\n",
      "18 1047\n",
      "18 1048\n",
      "18 1049\n",
      "139 1050\n",
      "139 1051\n",
      "139 1052\n",
      "139 1053\n",
      "139 1054\n",
      "139 1055\n",
      "139 1056\n",
      "139 1057\n",
      "139 1058\n",
      "139 1059\n",
      "139 1060\n",
      "25 1061\n",
      "18 1062\n",
      "18 1063\n",
      "18 1064\n",
      "18 1065\n",
      "18 1066\n",
      "25 1067\n",
      "25 1068\n",
      "25 1069\n",
      "25 1070\n",
      "25 1071\n",
      "25 1072\n",
      "26 1073\n",
      "26 1074\n",
      "26 1075\n",
      "18 1076\n",
      "18 1077\n",
      "18 1078\n",
      "18 1079\n",
      "18 1080\n",
      "18 1081\n",
      "18 1082\n",
      "25 1083\n",
      "25 1084\n",
      "25 1085\n",
      "25 1086\n",
      "25 1087\n",
      "25 1088\n",
      "25 1089\n",
      "25 1090\n",
      "26 1091\n",
      "26 1092\n",
      "26 1093\n",
      "26 1094\n",
      "26 1095\n",
      "26 1096\n",
      "26 1097\n",
      "26 1098\n",
      "26 1099\n",
      "140 1100\n",
      "140 1101\n",
      "140 1102\n",
      "140 1103\n",
      "140 1104\n",
      "140 1105\n",
      "140 1106\n",
      "140 1107\n",
      "141 1108\n",
      "141 1109\n",
      "141 1110\n",
      "141 1111\n",
      "141 1112\n",
      "141 1113\n",
      "141 1114\n",
      "141 1115\n",
      "141 1116\n",
      "141 1117\n",
      "141 1118\n",
      "142 1119\n",
      "142 1120\n",
      "142 1121\n",
      "142 1122\n",
      "142 1123\n",
      "143 1124\n",
      "143 1125\n",
      "143 1126\n",
      "143 1127\n",
      "144 1128\n",
      "144 1129\n",
      "144 1130\n",
      "145 1131\n",
      "145 1132\n",
      "145 1133\n",
      "12 1134\n",
      "12 1135\n",
      "12 1136\n",
      "12 1137\n",
      "144 1138\n",
      "144 1139\n",
      "144 1140\n",
      "12 1141\n",
      "12 1142\n",
      "47 1143\n",
      "135 1144\n",
      "72 1145\n",
      "144 1146\n",
      "144 1147\n",
      "144 1148\n",
      "144 1149\n",
      "144 1150\n",
      "144 1151\n",
      "144 1152\n",
      "144 1153\n",
      "144 1154\n",
      "144 1155\n",
      "144 1156\n",
      "144 1157\n",
      "144 1158\n",
      "144 1159\n",
      "144 1160\n",
      "144 1161\n",
      "144 1162\n",
      "144 1163\n",
      "97 1164\n",
      "97 1165\n",
      "97 1166\n",
      "97 1167\n",
      "97 1168\n",
      "97 1169\n",
      "97 1170\n",
      "95 1171\n",
      "95 1172\n",
      "115 1173\n",
      "95 1174\n",
      "95 1175\n",
      "95 1176\n",
      "95 1177\n",
      "95 1178\n",
      "97 1179\n",
      "105 1180\n",
      "146 1181\n",
      "144 1182\n",
      "144 1183\n",
      "144 1184\n",
      "115 1185\n",
      "124 1186\n",
      "124 1187\n",
      "124 1188\n",
      "124 1189\n",
      "124 1190\n",
      "124 1191\n",
      "124 1192\n",
      "124 1193\n",
      "146 1194\n",
      "146 1195\n",
      "147 1196\n",
      "147 1197\n",
      "115 1198\n",
      "115 1199\n",
      "115 1200\n",
      "148 1201\n",
      "2 1202\n",
      "2 1203\n",
      "149 1204\n",
      "150 1205\n",
      "150 1206\n",
      "151 1207\n",
      "151 1208\n",
      "2 1209\n",
      "2 1210\n",
      "2 1211\n",
      "2 1212\n",
      "2 1213\n",
      "2 1214\n",
      "2 1215\n",
      "2 1216\n",
      "2 1217\n",
      "52 1218\n",
      "52 1219\n",
      "52 1220\n",
      "124 1221\n",
      "124 1222\n",
      "124 1223\n",
      "124 1224\n",
      "152 1225\n",
      "152 1226\n",
      "152 1227\n",
      "152 1228\n",
      "152 1229\n",
      "152 1230\n",
      "152 1231\n",
      "152 1232\n",
      "152 1233\n",
      "152 1234\n",
      "152 1235\n",
      "152 1236\n",
      "153 1237\n",
      "154 1238\n",
      "154 1239\n",
      "154 1240\n",
      "17 1241\n",
      "155 1242\n",
      "156 1243\n",
      "156 1244\n",
      "156 1245\n",
      "156 1246\n",
      "156 1247\n",
      "156 1248\n",
      "156 1249\n",
      "156 1250\n",
      "156 1251\n",
      "156 1252\n",
      "156 1253\n",
      "156 1254\n",
      "156 1255\n",
      "156 1256\n",
      "156 1257\n",
      "156 1258\n",
      "156 1259\n",
      "157 1260\n",
      "157 1261\n",
      "158 1262\n",
      "158 1263\n",
      "159 1264\n",
      "159 1265\n",
      "159 1266\n",
      "159 1267\n",
      "159 1268\n",
      "159 1269\n",
      "159 1270\n",
      "76 1271\n",
      "76 1272\n",
      "160 1273\n",
      "160 1274\n",
      "160 1275\n",
      "160 1276\n",
      "160 1277\n",
      "160 1278\n",
      "160 1279\n",
      "160 1280\n",
      "160 1281\n",
      "161 1282\n",
      "145 1283\n",
      "162 1284\n",
      "155 1285\n",
      "155 1286\n",
      "161 1287\n",
      "13 1288\n",
      "42 1289\n",
      "42 1290\n",
      "163 1291\n",
      "164 1292\n",
      "164 1293\n",
      "164 1294\n",
      "165 1295\n",
      "165 1296\n",
      "166 1297\n",
      "167 1298\n",
      "99 1299\n",
      "99 1300\n",
      "168 1301\n",
      "168 1302\n",
      "85 1303\n",
      "85 1304\n",
      "85 1305\n",
      "85 1306\n",
      "85 1307\n",
      "85 1308\n",
      "85 1309\n",
      "85 1310\n",
      "85 1311\n",
      "85 1312\n",
      "85 1313\n",
      "85 1314\n",
      "85 1315\n",
      "164 1316\n",
      "164 1317\n",
      "164 1318\n",
      "164 1319\n",
      "164 1320\n",
      "164 1321\n",
      "164 1322\n",
      "164 1323\n",
      "164 1324\n",
      "164 1325\n",
      "164 1326\n",
      "164 1327\n",
      "164 1328\n",
      "164 1329\n",
      "166 1330\n",
      "99 1331\n",
      "99 1332\n",
      "165 1333\n",
      "165 1334\n",
      "165 1335\n",
      "165 1336\n",
      "165 1337\n",
      "165 1338\n",
      "165 1339\n",
      "165 1340\n",
      "165 1341\n",
      "165 1342\n",
      "85 1343\n",
      "169 1344\n",
      "169 1345\n",
      "169 1346\n",
      "170 1347\n",
      "169 1348\n",
      "169 1349\n",
      "171 1350\n",
      "64 1351\n",
      "76 1352\n",
      "76 1353\n",
      "80 1354\n",
      "85 1355\n",
      "85 1356\n",
      "85 1357\n",
      "85 1358\n",
      "85 1359\n",
      "85 1360\n",
      "85 1361\n",
      "85 1362\n",
      "85 1363\n",
      "85 1364\n",
      "85 1365\n",
      "12 1366\n",
      "13 1367\n",
      "13 1368\n",
      "13 1369\n",
      "13 1370\n",
      "13 1371\n",
      "13 1372\n",
      "12 1373\n",
      "12 1374\n",
      "12 1375\n",
      "172 1376\n",
      "172 1377\n",
      "172 1378\n",
      "172 1379\n",
      "172 1380\n",
      "172 1381\n",
      "172 1382\n",
      "172 1383\n",
      "172 1384\n",
      "172 1385\n",
      "173 1386\n",
      "174 1387\n",
      "175 1388\n",
      "27 1389\n",
      "27 1390\n",
      "27 1391\n",
      "27 1392\n",
      "27 1393\n",
      "27 1394\n",
      "27 1395\n",
      "27 1396\n",
      "27 1397\n",
      "27 1398\n",
      "27 1399\n",
      "27 1400\n",
      "27 1401\n",
      "27 1402\n",
      "27 1403\n",
      "27 1404\n",
      "27 1405\n",
      "27 1406\n",
      "27 1407\n",
      "27 1408\n",
      "176 1409\n",
      "25 1410\n",
      "177 1411\n",
      "177 1412\n",
      "27 1413\n",
      "27 1414\n",
      "27 1415\n",
      "27 1416\n",
      "27 1417\n",
      "177 1418\n",
      "177 1419\n",
      "177 1420\n",
      "177 1421\n",
      "12 1422\n",
      "12 1423\n",
      "12 1424\n",
      "12 1425\n",
      "12 1426\n",
      "12 1427\n",
      "12 1428\n",
      "123 1429\n",
      "123 1430\n",
      "123 1431\n",
      "123 1432\n",
      "123 1433\n",
      "123 1434\n",
      "123 1435\n",
      "123 1436\n",
      "123 1437\n",
      "123 1438\n",
      "123 1439\n",
      "123 1440\n",
      "178 1441\n",
      "178 1442\n",
      "178 1443\n",
      "178 1444\n",
      "178 1445\n",
      "178 1446\n",
      "178 1447\n",
      "178 1448\n",
      "178 1449\n",
      "179 1450\n",
      "179 1451\n",
      "179 1452\n",
      "179 1453\n",
      "179 1454\n",
      "179 1455\n",
      "179 1456\n",
      "179 1457\n",
      "177 1458\n",
      "177 1459\n",
      "180 1460\n",
      "180 1461\n",
      "180 1462\n",
      "180 1463\n",
      "180 1464\n",
      "180 1465\n",
      "180 1466\n",
      "180 1467\n",
      "180 1468\n",
      "123 1469\n",
      "123 1470\n",
      "123 1471\n",
      "123 1472\n",
      "123 1473\n",
      "123 1474\n",
      "123 1475\n",
      "123 1476\n",
      "123 1477\n",
      "123 1478\n",
      "123 1479\n",
      "123 1480\n",
      "123 1481\n",
      "123 1482\n",
      "123 1483\n",
      "123 1484\n",
      "123 1485\n",
      "123 1486\n",
      "123 1487\n",
      "123 1488\n",
      "123 1489\n",
      "123 1490\n",
      "123 1491\n",
      "123 1492\n",
      "123 1493\n",
      "123 1494\n",
      "181 1495\n",
      "181 1496\n",
      "181 1497\n",
      "182 1498\n",
      "120 1499\n",
      "123 1500\n",
      "123 1501\n",
      "123 1502\n",
      "123 1503\n",
      "123 1504\n",
      "123 1505\n",
      "123 1506\n",
      "178 1507\n",
      "178 1508\n",
      "178 1509\n",
      "178 1510\n",
      "178 1511\n",
      "179 1512\n",
      "179 1513\n",
      "179 1514\n",
      "179 1515\n",
      "179 1516\n",
      "179 1517\n",
      "179 1518\n",
      "181 1519\n",
      "181 1520\n",
      "181 1521\n",
      "181 1522\n",
      "180 1523\n",
      "180 1524\n",
      "183 1525\n",
      "184 1526\n",
      "184 1527\n",
      "184 1528\n",
      "184 1529\n",
      "184 1530\n",
      "184 1531\n",
      "184 1532\n",
      "184 1533\n",
      "79 1534\n",
      "79 1535\n",
      "79 1536\n",
      "79 1537\n",
      "79 1538\n",
      "79 1539\n",
      "79 1540\n",
      "79 1541\n",
      "79 1542\n",
      "79 1543\n",
      "79 1544\n",
      "79 1545\n",
      "79 1546\n",
      "185 1547\n",
      "185 1548\n",
      "185 1549\n",
      "186 1550\n",
      "186 1551\n",
      "186 1552\n",
      "186 1553\n",
      "186 1554\n",
      "186 1555\n",
      "80 1556\n",
      "80 1557\n",
      "80 1558\n",
      "80 1559\n",
      "185 1560\n",
      "185 1561\n",
      "185 1562\n",
      "185 1563\n",
      "185 1564\n",
      "185 1565\n",
      "185 1566\n",
      "186 1567\n",
      "186 1568\n",
      "186 1569\n",
      "186 1570\n",
      "186 1571\n",
      "186 1572\n",
      "186 1573\n",
      "184 1574\n",
      "187 1575\n",
      "73 1576\n",
      "184 1577\n",
      "184 1578\n",
      "24 1579\n",
      "24 1580\n",
      "24 1581\n",
      "24 1582\n",
      "24 1583\n",
      "24 1584\n",
      "24 1585\n",
      "24 1586\n",
      "24 1587\n",
      "24 1588\n",
      "24 1589\n",
      "24 1590\n",
      "24 1591\n",
      "24 1592\n",
      "24 1593\n",
      "24 1594\n",
      "24 1595\n",
      "24 1596\n",
      "24 1597\n",
      "24 1598\n",
      "24 1599\n",
      "24 1600\n",
      "24 1601\n",
      "24 1602\n",
      "24 1603\n",
      "24 1604\n",
      "24 1605\n",
      "24 1606\n",
      "24 1607\n",
      "24 1608\n",
      "24 1609\n",
      "24 1610\n",
      "24 1611\n",
      "24 1612\n",
      "24 1613\n",
      "24 1614\n",
      "24 1615\n",
      "24 1616\n",
      "188 1617\n",
      "188 1618\n",
      "188 1619\n",
      "188 1620\n",
      "189 1621\n",
      "189 1622\n",
      "189 1623\n",
      "189 1624\n",
      "190 1625\n",
      "190 1626\n",
      "190 1627\n",
      "191 1628\n",
      "191 1629\n",
      "191 1630\n",
      "191 1631\n",
      "191 1632\n",
      "191 1633\n",
      "183 1634\n",
      "192 1635\n",
      "193 1636\n",
      "193 1637\n",
      "193 1638\n",
      "193 1639\n",
      "193 1640\n",
      "193 1641\n",
      "194 1642\n",
      "195 1643\n",
      "195 1644\n",
      "196 1645\n",
      "79 1646\n",
      "30 1647\n",
      "30 1648\n",
      "30 1649\n",
      "197 1650\n",
      "197 1651\n",
      "197 1652\n",
      "197 1653\n",
      "197 1654\n",
      "197 1655\n",
      "198 1656\n",
      "198 1657\n",
      "198 1658\n",
      "198 1659\n",
      "198 1660\n",
      "30 1661\n",
      "30 1662\n",
      "122 1663\n",
      "122 1664\n",
      "122 1665\n",
      "122 1666\n",
      "122 1667\n",
      "122 1668\n",
      "122 1669\n",
      "122 1670\n",
      "122 1671\n",
      "196 1672\n",
      "196 1673\n",
      "196 1674\n",
      "122 1675\n",
      "122 1676\n",
      "12 1677\n",
      "12 1678\n",
      "12 1679\n",
      "12 1680\n",
      "12 1681\n",
      "12 1682\n",
      "12 1683\n",
      "199 1684\n",
      "199 1685\n",
      "199 1686\n",
      "199 1687\n",
      "199 1688\n",
      "199 1689\n",
      "199 1690\n",
      "199 1691\n",
      "199 1692\n",
      "199 1693\n",
      "199 1694\n",
      "199 1695\n",
      "199 1696\n",
      "199 1697\n",
      "199 1698\n",
      "199 1699\n",
      "199 1700\n",
      "199 1701\n",
      "199 1702\n",
      "199 1703\n",
      "12 1704\n",
      "12 1705\n",
      "12 1706\n",
      "12 1707\n",
      "199 1708\n",
      "199 1709\n",
      "199 1710\n",
      "199 1711\n",
      "199 1712\n",
      "199 1713\n",
      "199 1714\n",
      "199 1715\n",
      "199 1716\n",
      "12 1717\n",
      "12 1718\n",
      "12 1719\n",
      "12 1720\n",
      "12 1721\n"
     ]
    }
   ],
   "source": [
    "hostname = '127.0.0.1'\n",
    "username = 'alexacl95'\n",
    "password = 'SUSAna05'\n",
    "database = 'programacion'\n",
    "\n",
    "def doQuery( conn ) :\n",
    "    cur = conn.cursor()\n",
    "\n",
    "    cur.execute( \"select * from enfermedad_loci\" )\n",
    "\n",
    "    for id_nombre, nombre_enf in cur.fetchall() :\n",
    "        print (id_nombre, nombre_enf)\n",
    "\n",
    "myConnection = mysql.connector.connect( host=hostname, user=username, passwd=password, db=database )\n",
    "doQuery( myConnection )\n",
    "myConnection.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_diseaseId(disease_name):\n",
    "    cur = myConnection.cursor()\n",
    "\n",
    "    cur.execute( \"\"\"select * from enfermedad where nombre = \"%s\" \"\"\" % (disease_name) )\n",
    "    \n",
    "    id_enf = None\n",
    "    for id_, dissease in cur.fetchall() :\n",
    "        id_enf = id_\n",
    "        \n",
    "    if not id_enf:\n",
    "        #print(\"\"\"insert into enfermedad values (NULL, \"%s\" )\"\"\" % (disease_name))\n",
    "        cur.execute(\"\"\"insert into enfermedad values (NULL, \"%s\" )\"\"\" % (disease_name))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "        \n",
    "        id_enf = cur.fetchall()[0][0]\n",
    "        \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_enf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_platId(plat_name):\n",
    "    cur = myConnection.cursor()\n",
    "\n",
    "    cur.execute( \"\"\"select * from plataforma where nombre = \"%s\" \"\"\" % (plat_name) )\n",
    "    \n",
    "    id_plat = None\n",
    "    for id_, nombre_plat in cur.fetchall() :\n",
    "        id_plat = id_\n",
    "        \n",
    "    if not id_plat:\n",
    "        #print(\"\"\"insert into plataforma values (NULL, \"%s\" )\"\"\" % (plat_name))\n",
    "        cur.execute(\"\"\"insert into plataforma values (NULL, \"%s\" )\"\"\" % (plat_name))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "        \n",
    "                \n",
    "        id_plat = cur.fetchall()[0][0]\n",
    "        \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_plat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_jourId(jour_name):\n",
    "    cur = myConnection.cursor()\n",
    "\n",
    "    cur.execute( \"\"\"select * from journal where nombre = \"%s\" \"\"\" % (jour_name) )\n",
    "    \n",
    "    id_jour = None\n",
    "    for id_, nombre_jour in cur.fetchall() :\n",
    "        id_jour = id_\n",
    "        \n",
    "    if not id_jour:\n",
    "        #print(\"\"\"insert into journal values (NULL, \"%s\" )\"\"\" % (jour_name))\n",
    "        cur.execute(\"\"\"insert into journal values (NULL, \"%s\" )\"\"\" % (jour_name))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "        \n",
    "                \n",
    "        id_jour = cur.fetchall()[0][0]\n",
    "        \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_jour"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_estId(id_journal, id_enfermedad, est_name, tam_mues, replica):\n",
    "    cur = myConnection.cursor()\n",
    "\n",
    "    cur.execute( \"\"\"select * from estudio where nombre = \"%s\" \"\"\" % (est_name))\n",
    "    \n",
    "    id_est = None\n",
    "    for id_, id_journal, id_enfermedad, nombre_est, tam_mues, replica in cur.fetchall() :\n",
    "        id_est = id_\n",
    "        \n",
    "    if not id_est:\n",
    "        #print(\"\"\"insert into estudio values (NULL, \"%s\", \"%s\", \"%s\", \"%s\", \"%s\")\"\"\" % (id_journal, id_enfermedad,  est_name, tam_mues, replica))\n",
    "        cur.execute(\"\"\"insert into estudio values (NULL, \"%s\", \"%s\", \"%s\", \"%s\", \"%s\")\"\"\" % ( id_journal,id_enfermedad,est_name, tam_mues, replica))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "    \n",
    "               \n",
    "        id_est = cur.fetchall()[0][0]\n",
    "    \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_est"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_locId(id_plataforma, region, chrom, pos, genes_reportados, gen_mapped, gen_upstream,gen_downstream, SNP_GENE_ID, UPSTREAM_GENE_DISTANCE, DOWNSTREAM_GENE_DISTANCE, STRONGEST_SP_RISK, SNPS, MERGED, SNP_ID_CURRENT, CONTEXTO, risk_allele):\n",
    "    cur = myConnection.cursor()\n",
    "\n",
    "    cur.execute( \"\"\"select * from loci where gen_mapped = \"%s\" \"\"\" % (gen_mapped))\n",
    "    \n",
    "    id_loc = None\n",
    "    for id_,id_plataforma, region, chrom, pos, genes_reportados, gen_mapped, gen_upstream, gen_downstream, SNP_GENE_ID, UPSTREAM_GENE_DISTANCE, DOWNSTREAM_GENE_DISTANCE, STRONGEST_SP_RISK, SNPS, MERGED, SNP_ID_CURRENT, CONTEXTO, risk_allele in cur.fetchall() :\n",
    "        id_est = id_\n",
    "        \n",
    "    if not id_loc:\n",
    "        #print(\"\"\"insert into loci values (NULL, \"%s\",\"%s\", \"%s\", \"%s\", \"%s\",\n",
    "        #\"%s\",\"%s\", \"%s\", \"%s\", \"%s\",\n",
    "        #\"%s\",\"%s\", \"%s\", \"%s\", \"%s\",\n",
    "        #\"%s\",\"%s\")\"\"\" % (id_plataforma, region,chrom,pos,genes_reportados,\n",
    "        #                       gen_mapped, gen_upstream,gen_downstream, SNP_GENE_ID, \n",
    "        #                        UPSTREAM_GENE_DISTANCE, DOWNSTREAM_GENE_DISTANCE, \n",
    "        #                        STRONGEST_SP_RISK, SNPS, MERGED, SNP_ID_CURRENT, \n",
    "        #                        CONTEXTO, risk_allele))\n",
    "        \n",
    "        cur.execute(\"\"\"insert into loci values (NULL, \"%s\",\"%s\", \"%s\", \"%s\", \"%s\",\n",
    "        \"%s\",\"%s\", \"%s\", \"%s\", \"%s\",\n",
    "        \"%s\",\"%s\", \"%s\", \"%s\", \"%s\",\n",
    "        \"%s\",\"%s\")\"\"\" % (id_plataforma, region,chrom,pos,genes_reportados,\n",
    "                                gen_mapped, gen_upstream,gen_downstream, SNP_GENE_ID, \n",
    "                                UPSTREAM_GENE_DISTANCE, DOWNSTREAM_GENE_DISTANCE, \n",
    "                                STRONGEST_SP_RISK, SNPS, MERGED, SNP_ID_CURRENT, \n",
    "                                CONTEXTO, risk_allele))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "    \n",
    "               \n",
    "        id_loc = cur.fetchall()[0][0]\n",
    "    \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_loc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def get_pubId(id_journal, id_estudio, id_pubmed, autor, fecha_pub, link):\n",
    "    cur = myConnection.cursor()\n",
    "\n",
    "    cur.execute( \"\"\"select * from publicacion where id_pubmed = \"%s\" \"\"\" % (id_pubmed))\n",
    "    \n",
    "    id_pub = None\n",
    "    for id_, id_journal, id_estudio, idpubmed, autor, fecha_pub, link in cur.fetchall() :\n",
    "        id_pub = id_\n",
    "        \n",
    "    if not id_pub:\n",
    "        #print(\"\"\"insert into publicacion values (NULL, \"%s\", \"%s\", \"%s\", \"%s\", \"%s\", \"%s\")\"\"\" % (id_journal, id_estudio, id_pubmed, autor, fecha_pub, link))\n",
    "        cur.execute(\"\"\"insert into publicacion values (NULL, \"%s\", \"%s\", \"%s\", \"%s\", \"%s\", \"%s\")\"\"\" % (id_journal, id_estudio, id_pubmed, autor, fecha_pub, link))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "    \n",
    "               \n",
    "        id_pub = cur.fetchall()[0][0]\n",
    "    \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_pub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "def get_enfloId(loci_id, id_enfermedad,):\n",
    "    cur=myConnection.cursor()\n",
    "    \n",
    "    cur.execute( \"\"\"select * from enfermedad_loci where id_loci = \"%s\" \"\"\" % (id_loci))\n",
    "    \n",
    "    id_enflo = None\n",
    "    for id_, loci_id, id_enfermedad in cur.fetchall() :\n",
    "        id_enflo = id_\n",
    "        \n",
    "    if not id_enflo:    \n",
    "        print(\"\"\"insert into enfermedad_loci values (\"%s\",\"%s\")\"\"\" % (loci_id, id_enfermedad))\n",
    "        cur.execute(\"\"\"insert into enfermedad_loci values (\"%s\", \"%s\")\"\"\" % (loci_id, id_enfermedad))\n",
    "        cur.execute(\"SELECT LAST_INSERT_ID()\")\n",
    "        \n",
    "        id_enflo = cur.fetchall()[0][0]\n",
    "    \n",
    "        myConnection.commit()\n",
    "        \n",
    "    return id_enflo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "DataError",
     "evalue": "1406 (22001): Data too long for column 'tamano_muestra' at row 1",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mDataError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-25ff1f4b4f49>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     62\u001b[0m     \u001b[0mtam_mues\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mrow\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'INITIAL SAMPLE SIZE'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m     \u001b[0mreplica\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mrow\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'REPLICATION SAMPLE SIZE'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m---> 64\u001b[0;31m     \u001b[0mest_id\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mget_estId\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mid_journal\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mid_enfermedad\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mest_name\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtam_mues\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreplica\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     65\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m     66\u001b[0m     \u001b[1;31m#identidad publicacion\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-7-2c5e3b48034e>\u001b[0m in \u001b[0;36mget_estId\u001b[0;34m(id_journal, id_enfermedad, est_name, tam_mues, replica)\u001b[0m\n\u001b[1;32m     10\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0mid_est\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m         \u001b[1;31m#print(\"\"\"insert into estudio values (NULL, \"%s\", \"%s\", \"%s\", \"%s\", \"%s\")\"\"\" % (id_journal, id_enfermedad,  est_name, tam_mues, replica))\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m         \u001b[0mcur\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"\"\"insert into estudio values (NULL, \"%s\", \"%s\", \"%s\", \"%s\", \"%s\")\"\"\"\u001b[0m \u001b[1;33m%\u001b[0m \u001b[1;33m(\u001b[0m \u001b[0mid_journal\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mid_enfermedad\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mest_name\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtam_mues\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreplica\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m         \u001b[0mcur\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mexecute\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"SELECT LAST_INSERT_ID()\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\Alex\\Anaconda3\\lib\\site-packages\\mysql\\connector\\cursor.py\u001b[0m in \u001b[0;36mexecute\u001b[0;34m(self, operation, params, multi)\u001b[0m\n\u001b[1;32m    505\u001b[0m             \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_executed\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mstmt\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    506\u001b[0m             \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 507\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_handle_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_connection\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcmd_query\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mstmt\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    508\u001b[0m             \u001b[1;32mexcept\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mInterfaceError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    509\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_connection\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_have_next_result\u001b[0m\u001b[1;33m:\u001b[0m  \u001b[1;31m# pylint: disable=W0212\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\Alex\\Anaconda3\\lib\\site-packages\\mysql\\connector\\connection.py\u001b[0m in \u001b[0;36mcmd_query\u001b[0;34m(self, query)\u001b[0m\n\u001b[1;32m    720\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[0misinstance\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mquery\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mbytes\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    721\u001b[0m             \u001b[0mquery\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mquery\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mencode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'utf-8'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 722\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_handle_result\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_send_cmd\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mServerCmd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mQUERY\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mquery\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    723\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    724\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_have_next_result\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mC:\\Users\\Alex\\Anaconda3\\lib\\site-packages\\mysql\\connector\\connection.py\u001b[0m in \u001b[0;36m_handle_result\u001b[0;34m(self, packet)\u001b[0m\n\u001b[1;32m    638\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_handle_eof\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpacket\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    639\u001b[0m         \u001b[1;32melif\u001b[0m \u001b[0mpacket\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m4\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m255\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m--> 640\u001b[0;31m             \u001b[1;32mraise\u001b[0m \u001b[0merrors\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_exception\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpacket\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    641\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m    642\u001b[0m         \u001b[1;31m# We have a text result set\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mDataError\u001b[0m: 1406 (22001): Data too long for column 'tamano_muestra' at row 1"
     ]
    }
   ],
   "source": [
    "hostname = '127.0.0.1'\n",
    "username = 'alexacl95'\n",
    "password = 'SUSAna05'\n",
    "database = 'programacion'\n",
    "\n",
    "myConnection = mysql.connector.connect(host=hostname, user=username, passwd=password, db=database )\n",
    "cur=myConnection.cursor()\n",
    "\n",
    "for index, row in df.iterrows():\n",
    "    #identidad enfermedad\n",
    "    dissease_name = row['DISEASE/TRAIT']\n",
    "    dissease_id = get_diseaseId(dissease_name)\n",
    "        \n",
    "    #identidad plataforma\n",
    "    plat_name = row['PLATFORM [SNPS PASSING QC]']\n",
    "    #plat_id = get_platId(plat_name)\n",
    "    \n",
    "    #identidad loci\n",
    "    id_plataforma = plat_id\n",
    "    region=row['REGION']\n",
    "    chrom=row['CHR_ID']\n",
    "    pos=row['CHR_POS']\n",
    "    genes_reportados=row['REPORTED GENE(S)']\n",
    "    gen_mapped=row['MAPPED_GENE']\n",
    "    gen_upstream=row['UPSTREAM_GENE_ID']\n",
    "    gen_downstream=row['DOWNSTREAM_GENE_ID']\n",
    "    SNP_GENE_ID=row['SNP_GENE_IDS']\n",
    "    UPSTREAM_GENE_DISTANCE=row['UPSTREAM_GENE_DISTANCE']\n",
    "    DOWNSTREAM_GENE_DISTANCE=row['DOWNSTREAM_GENE_DISTANCE']\n",
    "    STRONGEST_SP_RISK=row['STRONGEST SNP-RISK ALLELE']\n",
    "    SNPS=row['SNPS']\n",
    "    MERGED=row['MERGED']\n",
    "    SNP_ID_CURRENT=row['SNP_ID_CURRENT']\n",
    "    CONTEXTO=row['CONTEXT']\n",
    "    risk_allele=row['RISK ALLELE FREQUENCY']\n",
    "    #PVAL=row['P-VALUE']\n",
    "    #Pvalue_MLOG=row['PVALUE_MLOG']\n",
    "    #PVALUE_txt=row['P-VALUE (TEXT)']\n",
    "    #BETA=row['OR or BETA']\n",
    "    #novCI= row['95% CI (TEXT)']\n",
    "    #id_loci= get_locId(id_plataforma, region,\n",
    "                       #chrom,pos,genes_reportados, \n",
    "                      # gen_mapped, gen_upstream,gen_downstream,\n",
    "                       #SNP_GENE_ID, UPSTREAM_GENE_DISTANCE,\n",
    "                       #DOWNSTREAM_GENE_DISTANCE,\n",
    "                       #STRONGEST_SP_RISK, SNPS, MERGED, \n",
    "                        #SNP_ID_CURRENT, CONTEXTO, risk_allele)\n",
    "   \n",
    "    #identidad loci enfermedad\n",
    "    id_enfermedad=dissease_id \n",
    "    loci_id=id_loci    \n",
    "    #enflo_id = get_enfloId(id_enfermedad, loci_id)\n",
    "    \n",
    "    #identidad journal\n",
    "    jour_name = row['JOURNAL']\n",
    "    jour_id = get_jourId(jour_name)\n",
    "    \n",
    "    #identidad estudio\n",
    "    id_enfermedad=dissease_id\n",
    "    id_journal=jour_id\n",
    "    est_name =  row['STUDY']\n",
    "    tam_mues = row['INITIAL SAMPLE SIZE']\n",
    "    replica = row['REPLICATION SAMPLE SIZE']\n",
    "    est_id = get_estId(id_journal, id_enfermedad, est_name, tam_mues, replica)\n",
    "    \n",
    "    #identidad publicacion\n",
    "    id_journal=jour_id\n",
    "    id_estudio=est_id\n",
    "    idpubmed= row['PUBMEDID']\n",
    "    autor= row['FIRST AUTHOR']\n",
    "    fecha_pub= row['DATE ADDED TO CATALOG']\n",
    "    link= row['LINK']\n",
    "    id_publicacion = get_pubId(id_journal, id_estudio, idpubmed, autor, fecha_pub, link)\n",
    "    \n",
    "                  \n",
    "myConnection.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ejercicio 3\n",
    "\n",
    "Realize de la base de datos una consulta que le responda una pregunta biológica\n",
    "(e.g. qué genes estan relacionados con cuales enfermedades)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***Enfermedad escogida: Asthma\n",
      "- Codigo enfermedad: \n",
      "73\n",
      "- Codigo loci:\n",
      "25\n",
      "- Cromosoma afectado:\n",
      "1\n",
      "- Region afectada:\n",
      "1q23.2\n",
      "- Posicion afectada:\n",
      "159711078\n",
      "\n",
      "***Enfermedad escogida: Allergic rhinitis\n",
      "- Codigo enfermedad: \n",
      "290\n",
      "- Codigo loci:\n",
      "83\n",
      "- Cromosoma afectado:\n",
      "3\n",
      "- Region afectada:\n",
      "3q23\n",
      "- Posicion afectada:\n",
      "141386728\n",
      "\n",
      "***Enfermedad escogida: Allergic rhinitis in asthma\n",
      "- Codigo enfermedad: \n",
      "719\n",
      "- Codigo loci:\n",
      "17\n",
      "- Cromosoma afectado:\n",
      "15\n",
      "- Region afectada:\n",
      "15q25.1\n",
      "- Posicion afectada:\n",
      "78513681\n",
      "\n",
      "***Enfermedad escogida: Atopic dermatitis\n",
      "- Codigo enfermedad: \n",
      "173\n",
      "- Codigo loci:\n",
      "60\n",
      "- Cromosoma afectado:\n",
      "11\n",
      "- Region afectada:\n",
      "11p15.4\n",
      "- Posicion afectada:\n",
      "5285279\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#cuales son los cromosomas, regiones y posiciones afectadas para determinada enfermedad\n",
    "\n",
    "hostname = '127.0.0.1'\n",
    "username = 'alexacl95'\n",
    "password = 'SUSAna05'\n",
    "database = 'programacion'\n",
    "\n",
    "myConnection = mysql.connector.connect(host=hostname, user=username, passwd=password, db=database )\n",
    "\n",
    "\n",
    "def enfermedades_info_1(nombres_enfermedades):\n",
    "    cur=myConnection.cursor()\n",
    "   \n",
    "    for i in range(len(nombres_enfermedades)):\n",
    "        cur.execute(\"\"\"select * FROM enfermedad where nombre= \"%s\" \"\"\" % nombres_enfermedades[i])\n",
    "        for id_, nombre_enf in cur.fetchall() :\n",
    "                print (\"***Enfermedad escogida: \" + nombre_enf)\n",
    "                print (\"- Codigo enfermedad: \" )\n",
    "                print (id_)\n",
    "                \n",
    "        cur.execute(\"\"\"select * FROM enfermedad_loci where id_enfermedad= \"%s\" \"\"\"  % id_)\n",
    "        for id_2, cod_enf in cur.fetchall() :\n",
    "                print (\"- Codigo loci:\")\n",
    "                print (id_2)\n",
    "               \n",
    "\n",
    "        cur.execute(\"\"\"select * FROM loci where id_loci= \"%s\" \"\"\" % id_2)\n",
    "        for id_3, id_plataforma, region, chrom, pos, genes_reportados, gen_mapped, gen_upstream,gen_downstream, SNP_GENE_ID, UPSTREAM_GENE_DISTANCE, DOWNSTREAM_GENE_DISTANCE, STRONGEST_SP_RISK, SNPS, MERGED, SNP_ID_CURRENT, CONTEXTO, risk_allele in cur.fetchall() :\n",
    "                print (\"- Cromosoma afectado:\")\n",
    "                print (chrom)\n",
    "                print (\"- Region afectada:\")\n",
    "                print (region) \n",
    "                print (\"- Posicion afectada:\")\n",
    "                print (pos)\n",
    "                print()\n",
    "                    \n",
    "      \n",
    "    return lol\n",
    "\n",
    "enfermedades=(\"Asthma\", \"Allergic rhinitis\", \n",
    "              \"Allergic rhinitis in asthma\",\n",
    "             \"Atopic dermatitis\")\n",
    "enfermedades_info_1(enfermedades)\n",
    "myConnection.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ejercicio 4\n",
    "\n",
    "Guarde el resultado de la consulta anterior en un archivo csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'enfermedades_c.csv'"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hostname = '127.0.0.1'\n",
    "username = 'alexacl95'\n",
    "password = 'SUSAna05'\n",
    "database = 'programacion'\n",
    "\n",
    "myConnection = mysql.connector.connect(host=hostname, user=username, passwd=password, db=database )\n",
    "\n",
    "def enfermedades_info_2(nombres_enfermedades):\n",
    "    cur=myConnection.cursor()\n",
    "    N= len(nombres_enfermedades)+1\n",
    "    lol = [[] for _ in range(N)]\n",
    "    lol[0].append(\"enfermedades\")\n",
    "    lol[0].append(\"Codigo enfermedad\") \n",
    "    lol[0].append(\"Codigo loci\")\n",
    "    lol[0].append(\"Cromosoma afectado\")\n",
    "    lol[0].append(\"Region afectada\")  \n",
    "    lol[0].append(\"posicion afectada\") \n",
    "    for i in range(len(nombres_enfermedades)):\n",
    "        cur.execute(\"\"\"select * FROM enfermedad where nombre= \"%s\" \"\"\" % nombres_enfermedades[i])\n",
    "        for id_, nombre_enf in cur.fetchall() :\n",
    "                lol[i+1].append(nombre_enf)\n",
    "        \n",
    "        cur.execute(\"\"\"select * FROM enfermedad_loci where id_enfermedad= \"%s\" \"\"\"  % id_)\n",
    "        for id_2, cod_enf in cur.fetchall() :\n",
    "                lol[i+1].append(id_2)\n",
    "\n",
    "        cur.execute(\"\"\"select * FROM loci where id_loci= \"%s\" \"\"\" % id_2)\n",
    "        for id_3, id_plataforma, region, chrom, pos, genes_reportados, gen_mapped, gen_upstream,gen_downstream, SNP_GENE_ID, UPSTREAM_GENE_DISTANCE, DOWNSTREAM_GENE_DISTANCE, STRONGEST_SP_RISK, SNPS, MERGED, SNP_ID_CURRENT, CONTEXTO, risk_allele in cur.fetchall() :\n",
    "                lol[i+1].append(id_3)\n",
    "                lol[i+1].append(chrom)\n",
    "                lol[i+1].append(region)\n",
    "                lol[i+1].append(pos)      \n",
    "       \n",
    "       \n",
    "        import csv\n",
    "        with open('enfermedades_c.csv', 'w', newline='') as csvfile:\n",
    "            writer = csv.writer(csvfile)\n",
    "            writer.writerows(lol)\n",
    "                  \n",
    "       \n",
    "    return 'enfermedades_c.csv'\n",
    "\n",
    "enfermedades=(\"Asthma\", \"Allergic rhinitis\", \n",
    "              \"Allergic rhinitis in asthma\",\n",
    "             \"Atopic dermatitis\")\n",
    "enfermedades_info_2(enfermedades)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
